src/Reflection/ReflectionParameter.php

```diff
@@ -208,8 +209,20 @@ private function parseDefaultValueNode() : void
         $defaultValueNode = $this->node->default;
 
         if ($defaultValueNode instanceof Node\Expr\ClassConstFetch) {
+            $className = $this->getRealClassName(implode('\\', $defaultValueNode->class->parts));
+
+            if ($className instanceof Self_) {
+                $class = $this->function->getDeclaringClass();
+                do {
+                    if ($class->hasConstant($defaultValueNode->name)) {
+                        $className = $class->getName();
+                        break;
+                    }
+                } while ($class = $class->getParentClass());
+            }
+
             $this->isDefaultValueConstant = true;
-            $this->defaultValueConstantName = $defaultValueNode->name;
+            $this->defaultValueConstantName = ltrim($className, '\\') . '::' . $defaultValueNode->name;
```

Comment:
So we are casting to string and prepending `\\` ourselves? That's a bug that we should solve in a separate issue then :-\